src/renderers/dom/shared/dangerousStyleValue.js

```diff
@@ -48,7 +51,27 @@ function dangerousStyleValue(name, value) {
   }
 
   if (typeof value === 'string') {
-    value = value.trim();
+    if (__DEV__) {
+      if (component) {
+        var owner = component._currentElement._owner;
+        var componentName = owner ? owner.getName() : component._currentElement.type;
+        if (!styleWarnings[componentName]) {
+          styleWarnings[componentName] = {};
+        }
+        var warnings = styleWarnings[componentName];
+        if (!warnings[name]) {
+          warnings[name] = true;
+          warning(
+            false,
+            '`%s` passed a numeric string value for CSS property `%s` ' +
+            'which will be treated as a unitless number.',
+            componentName,
+            name
+          );
+        }
+      }
+    }
+    return value.trim();
```

Comment:
We usually warn for a release without a functionality change, and then change functionality the release after.  The reason being: if we are emitting warnings, users probably won't/shouldn't be using the "feature" that we are warning for (ie. they need to avoid the feature in order to avoid the warning).  For this reason, users won't benefit from the functionality change this release, so we might as well forestall the functionality change until they would actually be able to use it (so as to avoid breaking their apps when they update).